carbon-design-system/carbon-addons-iot-react
# refactor(date-time-picker): sync V1 and V2, refactor a lot of common code, sync tests

Closes #3117 

**Summary**

- refactors a lot of shared code from both DateTimePickers and turns it into shared functions and hooks
- adds tests to confirm new behaviors added to either
- synced tests so both versions are fully tested

**Change List (commits, features, bugs, etc)**

- upgraded cypress due to weird test failures
- added keyboard navigation (and tests) to V2
- added time validation to V2
- added tests to confirm time validation to both v1 and v2
- refactored lots of shared code into shared functions and hooks for both v1 and v2

**Acceptance Test (how to verify the PR)**

- confirm DateTImePicker V1 and V2 both still function as expected

**Regression Test (how to make sure this PR doesn't break old functionality)**

- check Card with DateTimePicker to confirm no adverse effects to v2



**Things to look for during review**
